api/src/test/java/org/pmiops/workbench/api/NotebooksControllerTest.java

```diff
@@ -118,522 +104,188 @@ class MockNotebook {
-
-    stubGetWorkspace(fromWorkspace, WorkspaceAccessLevel.OWNER);
-    stubGetWorkspace(toWorkspace, WorkspaceAccessLevel.OWNER);
-
-    assertThrows(
-        BadRequestException.class,
-        () ->
-            notebooksController.copyNotebook(
-                fromWorkspace.getWorkspaceNamespace(),
-                fromWorkspace.getFirecloudName(),
-                fromNotebookName,
-                new CopyRequest()
-                    .toWorkspaceName(toWorkspace.getFirecloudName())
-                    .toWorkspaceNamespace(toWorkspace.getWorkspaceNamespace())
-                    .newName(newNotebookName)));
-  }
+    when(mockNotebookService.copyNotebook(
+            anyString(), anyString(), anyString(), anyString(), anyString(), anyString()))
+        .thenReturn(expectedFileDetail);
 
-  @Test
-  public void testCloneNotebook() {
-    DbWorkspace workspace = createWorkspace();
-    stubGetWorkspace(workspace, WorkspaceAccessLevel.OWNER);
-
-    String nb1 = NotebooksService.withNotebookExtension("notebooks/nb1");
-    String newPath = NotebooksService.withNotebookExtension("notebooks/Duplicate of nb1");
-    String fullPath = BUCKET_URI + newPath;
-    notebooksController.cloneNotebook(
-        workspace.getWorkspaceNamespace(),
-        workspace.getFirecloudName(),
-        NotebooksService.withNotebookExtension("nb1"));
-    verify(mockCloudStorageClient)
-        .copyBlob(
-            BlobId.of(TestMockFactory.WORKSPACE_BUCKET_NAME, nb1),
-            BlobId.of(TestMockFactory.WORKSPACE_BUCKET_NAME, newPath));
-    verify(mockUserRecentResourceService)
-        .updateNotebookEntry(eq(workspace.getWorkspaceId()), anyLong(), eq(fullPath));
-  }
+    FileDetail actualFileDetail =
+        notebooksController
+            .copyNotebook(
+                FROM_WORKSPACE_NAMESPACE, FROM_WORKSPACE_NAME, FROM_NOTEBOOK_NAME, copyRequest)
+            .getBody();
 
-  @Test
-  public void testDeleteNotebook() {
-    DbWorkspace workspace = createWorkspace();
-    stubGetWorkspace(workspace, WorkspaceAccessLevel.OWNER);
-
-    String nb1 = NotebooksService.withNotebookExtension("notebooks/nb1");
-    String fullPath = BUCKET_URI + nb1;
-    notebooksController.deleteNotebook(
-        workspace.getWorkspaceNamespace(),
-        workspace.getFirecloudName(),
-        NotebooksService.withNotebookExtension("nb1"));
-    verify(mockCloudStorageClient)
-        .deleteBlob(BlobId.of(TestMockFactory.WORKSPACE_BUCKET_NAME, nb1));
-    verify(mockUserRecentResourceService)
-        .deleteNotebookEntry(eq(workspace.getWorkspaceId()), anyLong(), eq(fullPath));
-  }
+    verify(mockNotebookService)
+        .copyNotebook(
+            FROM_WORKSPACE_NAMESPACE,
+            FROM_WORKSPACE_NAME,
+            FROM_NOTEBOOK_NAME,
+            toWorkspaceNamespace,
+            toWorkspaceName,
+            TO_NOTEBOOK_NAME);
 
-  @Test
-  public void testDeleteNotebook_writer() {
```

Comment:
these 2 can be deleted if a call to check workspace auth is added to the new service test - but this check is not yet present

Reply:
I believe that this line in NotebookServiceTest's testDeleteNotebook test is what you are referring to:
```
    verify(mockWorkspaceAuthService)
        .enforceWorkspaceAccessLevel(NAMESPACE_NAME, WORKSPACE_NAME, WorkspaceAccessLevel.WRITER);
```

Not sure if I added this after your comment or not.